### PR TITLE
add devcontainer for codespaces

### PR DESCRIPTION
**Pull request by rbtr**
Signed-off-by: GitHub <noreply@github.com>

**Reason for Change**:

adds trimmed down (go+git+k8s) config for codespaces 

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

**Notes**:

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+			"VARIANT": "1-bullseye",
+			// Options
+			"NODE_VERSION": "none"
+		}
+	},
+	"runArgs": [
+		"--init",
+		"--cap-add=SYS_PTRACE",
+		"--security-opt",
+		"seccomp=unconfined"
```
Comment (nitishm) [inline .devcontainer/devcontainer.json:20]: Out of curiousity are these required/being used?

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	"extensions": [
+		"golang.Go"
+	],
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	// "forwardPorts": [],
+	// Use 'postCreateCommand' to run commands after the container is created.
+	"postCreateCommand": "make setup",
+	// Comment out connect as root instead. More info: https://aka.ms/vscode-remote/containers/non-root.
+	"remoteUser": "vscode",
+	"features": {
```
Comment (nitishm) [inline .devcontainer/devcontainer.json:39]: homebrew would be nice to have :)

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (rbtr) [reply .devcontainer/devcontainer.json:20]: required for dind, and we're the only container running on this vm so it's not a concern

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (rbtr) [reply .devcontainer/devcontainer.json:39]: i'm not against it, but i'm not convinced it adds enough value on linux earn a spot in the base container. you can install it in your codespace manually or via the dotfiles integration.

**Review by nitishm — APPROVED** [review]
LGTM

**Pull request merged**